gno.land/pkg/sdk/vm/keeper.go

```diff
@@ -217,16 +218,45 @@ func (vm *VMKeeper) Call(ctx sdk.Context, msg MsgCall) (res string, err error) {
 	mpn := gno.NewPackageNode("main", "main", nil)
 	mpn.Define("pkg", gno.TypedValue{T: &gno.PackageType{}, V: pv})
 	mpv := mpn.NewPackage()
-	// Parse expression.
+
+	if len(msg.Args) != len(ft.Params) {
+		return "", fmt.Errorf("wrong number of arguments in call to %s: want %d got %d", fnc, len(ft.Params), len(msg.Args))
+	}
+
+	request := make([]gno.TypedValue, len(ft.Params))
```

Comment:
I'm having trouble understanding the concept since `q_eval` is intended to evaluate GNO, not JSON. Im maybe wrong, but evaluating JSON seems out of place in this context. 
To clarify, I think the idea of evaluation can make sense, but when `q_eval` is used to evaluate GNO literally, it can lead to confusion when intending to evaluate an expression as JSON. 

A potentially sensible approach might be to allow the definition of structures/array in JSON using other parameters, which could then be used as variables within the expression. It will also make the implementation easier.

Reply:
You're right, the input doesn't make sense. Can we JSON marshal anything "returned" by `q_eval`, if the expression yields a result?